src/overlays/actors/ovl_Obj_Nozoki/z_obj_nozoki.c

```diff
@@ -37,54 +44,427 @@ const ActorInit Obj_Nozoki_InitVars = {
+            this->unk_15D = 1;
+            this->unk_15E = 70;
+        }
+    } else if (this->unk_15D == 1) {
+        if (D_80BA36B8 > 40.0f) {
+            this->unk_15D = 2;
+            Actor_PlaySfxAtPos(&this->dyna.actor, NA_SE_PL_SIT_ON_HORSE);
+        } else if (this->unk_15E != 0) {
+            if (DECR(this->unk_15E) == 0) {
+                Actor_PlaySfxAtPos(&this->dyna.actor, NA_SE_EV_CONVEYOR_SHUTTER_OPEN);
+            }
+        } else {
+            Math_StepToF(&this->dyna.actor.world.pos.y, this->dyna.actor.home.pos.y + 50.0f, 4.0f);
+        }
+    } else {
+        Math_StepToF(&this->dyna.actor.world.pos.y, this->dyna.actor.home.pos.y, 4.0f);
+    }
+}
+
+void func_80BA3230(ObjNozoki* this, GlobalContext* globalCtx) {
+    if (gSaveContext.save.weekEventReg[64] & 0x20) {
+        Actor* npc = globalCtx->actorCtx.actorLists[ACTORCAT_NPC].first;
+        Actor* test3 = SubS_FindActor(globalCtx, npc, ACTORCAT_NPC, ACTOR_EN_TEST3);
+
+        if ((test3 != NULL) && (test3->draw != NULL)) {
+            if ((globalCtx->curSpawn == 3) && !(gSaveContext.save.weekEventReg[64] & 0x40)) {
+                this->dyna.actor.flags |= (ACTOR_FLAG_1 | ACTOR_FLAG_8 | ACTOR_FLAG_10000);
+                this->dyna.actor.textId = 0x297A;
+            } else {
+                this->dyna.actor.flags |= (ACTOR_FLAG_1 | ACTOR_FLAG_8);
+                if (gSaveContext.save.weekEventReg[64] & 0x40) {
+                    this->dyna.actor.textId = 0;
+                } else {
+                    this->dyna.actor.textId = 0x2979;
+                }
+            }
+
+            if (Actor_ProcessTalkRequest(&this->dyna.actor, &globalCtx->state)) {
+                ObjNozoki_SetupAction(this, func_80BA3344);
+            } else {
+                func_800B8614(&this->dyna.actor, globalCtx, 50.0f);
+            }
+        }
+    }
+}
+
+void func_80BA3344(ObjNozoki* this, GlobalContext* globalCtx) {
+    if ((globalCtx->curSpawn == 3) && !(gSaveContext.save.weekEventReg[64] & 0x40)) {
+        if (Actor_TextboxIsClosing(&this->dyna.actor, globalCtx)) {
+            gSaveContext.save.weekEventReg[64] |= 0x40;
+            this->dyna.actor.flags &= ~ACTOR_FLAG_10000;
+            ObjNozoki_SetupAction(this, func_80BA3230);
+        }
+    } else if ((this->dyna.actor.textId == 0) || Actor_TextboxIsClosing(&this->dyna.actor, globalCtx)) {
+        globalCtx->nextEntranceIndex = 0xE20;
+        globalCtx->sceneLoadFlag = 0x14;
+    }
+}
+
+void ObjNozoki_Update(Actor* thisx, GlobalContext* globalCtx) {
+    ObjNozoki* this = THIS;
+
+    this->actionFunc(this, globalCtx);
+}
+
+Gfx* D_80BA34FC[] = {
+    object_secom_obj_DL_000080,
+    (Gfx*)0x0A0001A0,
+    object_secom_obj_DL_001230,
+    object_secom_obj_DL_001300,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Nozoki/ObjNozoki_Update.s")
+void func_80BA3434(Actor* thisx, GlobalContext* globalCtx) {
```

Comment:
This seems to be the main and only draw function, because it is the only one set to `actor.draw`, so I think it should be named accordingly
```suggestion
void ObjNozoki_Draw(Actor* thisx, GlobalContext* globalCtx) {
```